### ISSUE TITLE
Sorting in the Default UI

### ISSUE THREAD

**Issue by searls** [CloudCannon/pagefind#508]
Since filtering is supported in the Default UI, I was wondering if the maintainers are opening to adding support for sorting in the Default UI as well? I was surprised that to simply sort results in reverse-chronological order I would need to throw out the Default UI and roll my own, for instance.

The interface I'd like is:

```js
window.addEventListener('DOMContentLoaded', (event) => {
  new PagefindUI({
    element: "#search",
    resetStyles: false,
    sort: { date: "asc" }
  });
});
```

I imagine this would entail forwarding this option along all the way to [this line](https://github.com/CloudCannon/pagefind/blob/main/pagefind_ui/default/svelte/ui.svelte#L188), much in the same way `filters` is.

**Comment by bglw** [comment]
Yes, this is an easy enough option to pass through! I'll wire that up.

**Comment by searls** [comment]
That'd be incredible, thank you!On Nov 22, 2023, at 20:55, Liam Bigelow ***@***.***> wrote:﻿
Yes, this is an easy enough option to pass through! I'll wire that up.

**Comment by bglw** [comment]
Hey @searls — this is available for you to start using in a `1.0.5-rc0` release of Pagefind 🙂

**Comment by searls** [comment]
Awesome, thank you!

**Comment by searls** [comment]
I just tested this out and it's not working for me.

My markup includes date metadata and sort attributes:

```html
<time pubdate="" datetime="2010-12-07" data-pagefind-meta="date:2010-12-07" data-pagefind-sort="date:2010-12-07" class="block mb-2 text-sm font-semibold text-center text-secondary">
      Tuesday, Dec 7, 2010
</time>
```

My JS:

```js
  window.pageFindUi = new PagefindUI({
    element: "#search",
    resetStyles: false,
    autofocus: true,
    sort: { date: "desc" }
  });
```

But when I search, results are out-of-order:

![Screenshot 2023-11-29 at 9 15 31 AM](https://github.com/CloudCannon/pagefind/assets/79303/8b25eb83-96d9-48b8-9d84-1fec344c7198)

I deployed this so you can see a live version: [justin.searls.co/search](https://justin.searls.co/search/)

**Comment by bglw** [comment]
🤦 This is why I publish prereleases! My bad — sorry about that.

`v1.0.5-rc1` is out now 🙂

**Comment by searls** [comment]
Fixed! Looks good.

Also, you'll want to remove this line from the [sorting docs](https://unreleased.pagefind.app/docs/sorts/):

```
Sorting is currently only supported when interacting with the Pagefind JS API directly.
```

**Comment by volodymyrprokopyuk** [comment]
I hope that the official release v1.0.5 that includes sorting in the Default UI will happen soon. This feature is much needed. I use the standalone binary of pagefind from GitHub. Thank you!

**Comment by bglw** [comment]
Sorry for the delay all! This has now [landed in the v1.1.0 release](https://github.com/CloudCannon/pagefind/releases/tag/v1.1.0) :-)

**Comment by brouberol** [comment]
I'm just dropping by to say thank you! I've been hoping for this feature for years, and it's working like a charm <3
